### PR TITLE
KAFKA-9947; Ensure proper shutdown of components in `TransactionsBounceTest`

### PR DESCRIPTION
**Pull request by hachikuji**
This test case should ensure that clients and the bounce scheduler get shutdown properly. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala [hunk 1]
```diff
@@ -130,13 +129,10 @@ class TransactionsBounceTest extends KafkaServerTestHarness {
         iteration += 1
       }
     } finally {
-      producer.close()
-      consumer.close()
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala:134]: We are now relying on the lifecycle methods to close the producer and consumer. Did you do that to simplify the code, i.e. it has no impact on the behavior of the test?

Comment (hachikuji) [reply core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala:134]: Yeah, I thought it is a preferred pattern to simplify cleanup a little bit in test cases. It does not change the behavior of the test.

File: core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala [hunk 2]
```diff
@@ -156,17 +152,26 @@ class TransactionsBounceTest extends KafkaServerTestHarness {
...
+  }
 
-    verifyingConsumer.close()
+  private def createTransactionalProducer(transactionalId: String) = {
+    val props = new Properties()
+    props.put(ProducerConfig.ACKS_CONFIG, "all")
+    props.put(ProducerConfig.BATCH_SIZE_CONFIG, "512")
+    props.put(ProducerConfig.TRANSACTIONAL_ID_CONFIG, transactionalId)
+    props.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, "true")
+    createProducer(configOverrides = props)
```
Comment (ijuma) [inline core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala:163]: I was checking the difference between what we do now and before and noticed that we have this line in the helper method we were using previously:

> props.put(ProducerConfig.REQUEST_TIMEOUT_MS_CONFIG, deliveryTimeoutMs.toString)

That seems a bit questionable. Should we remove it from that method?

Comment (hachikuji) [reply core/src/test/scala/integration/kafka/api/TransactionsBounceTest.scala:163]: Nice catch. Yeah, let me fix that.

**Review by ijuma — APPROVED** [review]
LGTM if the tests pass. Let's please run the test where we changed the timeout in a loop locally to make sure it's not flaky.

**Comment by hachikuji** [comment]
I changed the patch to set the request timeout the same as the delivery timeout for this test case, which was the old behavior.

**Comment by ijuma** [comment]
Sounds good.

**Pull request merged**